api-report/shared-object-base.api.md

```diff
@@ -26,6 +26,11 @@ import { ITelemetryLoggerExt } from '@fluidframework/telemetry-utils';
 // @public
 export function createSingleBlobSummary(key: string, content: string | Uint8Array): ISummaryTreeWithStats;
 
+// @public
+export type FluidSerializableReadOnly = IFluidHandle | string | number | boolean | null | readonly FluidSerializableReadOnly[] | {
+    readonly [P in string]?: FluidSerializableReadOnly;
```

Comment:
What's the difference between this and something like `Readonly<Partial<Record<string, FluidSerializableReadOnly>>>`?  Not sure what the `[P in string]` is doing here.

Reply:
You would think that would be the same, but if you do your approach it fails to compile due to "circularly referencing itself".